packages/lib/test/timescape.test.ts

```diff
@@ -220,6 +220,43 @@ describe('timescape', () => {
       expect(years).toHaveFocus()
     })
 
+    it('should cycle through fields by Enter key', () => {
+      document.body.appendChild(container)
+      const { years, months, days, hours, minutes, seconds, ampm } = getFields()
+
+      years.focus()
+      expect(years).toHaveFocus()
+
+      const elements = [
+        months,
+        days,
+        hours,
+        minutes,
+        seconds,
+        ampm,
+        // start from beginning
+        years,
+      ]
+      for (const element of elements) {
+        fireEvent.keyDown(document.activeElement!, { key: 'Enter' })
+        expect(element).toHaveFocus()
+      }
+    })
+
+    it('should cycle through fields by the Tab key', () => {
+      document.body.appendChild(container)
+      const { years, months, days, hours, minutes, seconds, ampm } = getFields()
+
+      years.focus()
+      expect(years).toHaveFocus()
+
+      const elements = [months, days, hours, minutes, seconds, ampm]
```

Comment:
Since `fireEvent` is being used instead of `userEvent`, even if you add another element to the `container`, it wouldn't focus on the next element in this test environment.

Reply:
Huh interesting, I wasn't aware of `userEvent`, it looks like the tests should switch to using this instead of `fireEvent`

Reply:
Yeah, it's only really important when you need to emulate user behaviour. I'll probably be migrating TanStack Router over to it as well soon.